tools/functional/api/test_users.py

```diff
@@ -3,6 +3,158 @@
 https://docs.gitlab.com/ee/api/users.html
 https://docs.gitlab.com/ee/api/users.html#delete-authentication-identity-from-user
 """
+import time
+from pathlib import Path
+
+import pytest
+import requests
+
+
+@pytest.fixture(scope="session")
+def avatar_path(test_dir):
+    return test_dir / "fixtures" / "avatar.png"
+
+
+def test_create_user(gl, avatar_path):
+    user = gl.users.create(
+        {
+            "email": "foo@bar.com",
+            "username": "foo",
+            "name": "foo",
+            "password": "foo_password",
+            "avatar": open(avatar_path, "rb"),
+        }
+    )
+
+    created_user = gl.users.list(username="foo")[0]
+    assert created_user.username == user.username
+    assert created_user.email == user.email
+
+    avatar_url = user.avatar_url.replace("gitlab.test", "localhost:8080")
+    uploaded_avatar = requests.get(avatar_url).content
+    assert uploaded_avatar == open(avatar_path, "rb").read()
+
+
+def test_block_user(gl, user):
+    user.block()
+    users = gl.users.list(blocked=True)
+    assert user in users
+
+    user.unblock()
+    users = gl.users.list(blocked=False)
+    assert user in users
+
+
+def test_delete_user(gl, wait_for_sidekiq):
+    new_user = gl.users.create(
+        {
+            "email": "delete-user@test.com",
+            "username": "delete-user",
+            "name": "delete-user",
+            "password": "delete-user-pass",
+        }
+    )
+
+    new_user.delete()
+    wait_for_sidekiq()
+
+    assert new_user.id not in [user.id for user in gl.users.list()]
+
+
+def test_user_projects_list(gl, user):
+    projects = user.projects.list()
+    assert len(projects) == 0
```

Comment:
```suggestion
    assert projects
```

Reply:
These `len()` asserts are all evil, especially with group objects :D but maybe we could keep them as a sanity check as in "it _does_ return a list (rather than None) but the list it is empty"? Since None would fail with `len()`.

Or would you prefer the more explicit:
```python
assert isinstance(something, list)
assert not something
```